### PR TITLE
Frontend: add syntactic sugar for schedule and storage types.

### PR DESCRIPTION
**Pull request by orausch**
This enables parsing statements like
```python
for i in dace.map[0:N] @ ScheduleType.GPU_Device
  ...
```
And:
```python
def add(X: dace.float32[32] @ StorageType.GPU_Global):
  ...
```
(Inspired by https://exo-lang.dev)

### CODE REVIEW

File: dace/data.py [hunk 1]
```diff
@@ -321,6 +321,20 @@ def set_strides_from_layout(self,
         self.strides = strides
         self.total_size = totalsize
 
+    def __matmul__(self, storage: dtypes.StorageType):
+        """
+        Syntactic sugar for specifying the storage of a data descriptor.
+        This lets us write code like:
+
+        .. code-block:: python
+            @dace
```
Comment (tbennun) [inline dace/data.py:330]:
```suggestion

            @dace
```

File: dace/data.py [hunk 2]
```diff
@@ -321,6 +321,20 @@ def set_strides_from_layout(self,
         self.strides = strides
         self.total_size = totalsize
 
+    def __matmul__(self, storage: dtypes.StorageType):
+        """
+        Syntactic sugar for specifying the storage of a data descriptor.
+        This lets us write code like:
```
Comment (tbennun) [inline dace/data.py:327]:
```suggestion
        This enables controlling the storage location as follows:
```

File: tests/python_frontend/device_annotations_test.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import dace
```
Comment (tbennun) [inline tests/python_frontend/device_annotations_test.py:1]:
```suggestion
# Copyright 2019-2022 ETH Zurich and the DaCe authors. All rights reserved.
import dace
```

**Review by tbennun — CHANGES_REQUESTED** [review]
Cool PR!

**Comment by orausch** [comment]
You're going to have to approve for auto-merge :)

**Comment by gronerl** [comment]
I think this breaks running non-decorated numpy code for the maps, right? Would it make sense to also add a `__matmul__` to `dace.map[]` ?

**Review by tbennun — APPROVED** [review]

**Comment by tbennun** [comment]
> I think this breaks running non-decorated numpy code for the maps, right? Would it make sense to also add a `__matmul__` to `dace.map[]` ?

Thank you! Done.

**Pull request merged**